engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/error/ErrorEventTest.java

```diff
@@ -707,4 +707,86 @@ public void shouldPropagateErrorMessageVariable() {
             .getFirst();
     Assertions.assertThat(variableRecords.getValue()).hasValue("\"error-message\"");
   }
+
+  @Test
+  public void shouldPropagateEmptyErrorMessageVariable() {
+    // Given
+    final var process =
+        Bpmn.createExecutableProcess(PROCESS_ID)
+            .startEvent()
+            .serviceTask("task", t -> t.zeebeJobType(JOB_TYPE))
+            .boundaryEvent(
+                "error-boundary-event",
+                b ->
+                    b.errorEventDefinition().error(ERROR_CODE).errorMessageVariable("errorMessage"))
+            .endEvent()
+            .done();
+
+    ENGINE.deployment().withXmlResource(process).deploy();
+
+    final var processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    // when
+    ENGINE
+        .job()
+        .ofInstance(processInstanceKey)
+        .withType(JOB_TYPE)
+        .withErrorCode(ERROR_CODE)
+        .throwError();
+
+    // Then
+    final long scopeKey =
+        RecordingExporter.processInstanceRecords()
+            .withProcessInstanceKey(processInstanceKey)
+            .withElementType(BpmnElementType.BOUNDARY_EVENT)
+            .getFirst()
+            .getKey();
+
+    final Record<VariableRecordValue> variableRecords =
+        RecordingExporter.variableRecords()
+            .withProcessInstanceKey(processInstanceKey)
+            .withScopeKey(scopeKey)
+            .withName("errorMessage")
+            .getFirst();
+    Assertions.assertThat(variableRecords.getValue()).hasValue("\"\"");
```

Comment:
🔧 Let's add a description to the assertions to clarify why we're expecting this:

```java
Assertions.assertThat(variableRecords.getValue())
  .describedAs("Expect that errorMessage variable has an empty string as value because no error message was provided with the thrown error")
  .hasValue("\"\"");
```

🔧 We can do the same with the test case below.